From e06f291cc36c9e9a3c9c224e23109ac6dad4eb1c Mon Sep 17 00:00:00 2001
From: Heiko Thiery <heiko.thiery@gmail.com>
Date: Thu, 25 Nov 2021 12:29:47 +0100
Subject: [PATCH] libnetconf2 BUGFIX netopeer2 build fails when building with
 musl libc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/home/hthiery/sources/mainline/buildroot/output/host/x86_64-buildroot-linux-musl/sysroot/usr/include/libnetconf2/session_server.h:439:55: error: unknown type name ‘mode_t’
  439 | int nc_server_endpt_set_perms(const char *endpt_name, mode_t mode, uid_t uid, gid_t gid);

For musl libc the include "sys/types.h" is required.

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>
---
 src/session_server.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/session_server.h b/src/session_server.h
index 4507eea..4779119 100644
--- a/src/session_server.h
+++ b/src/session_server.h
@@ -19,6 +19,7 @@
 extern "C" {
 #endif
 
+#include <sys/types.h>
 #include <libyang/libyang.h>
 #include <stdint.h>
 
-- 
2.30.2

